Release SciBib version 2

benibaeumle opened the pull request.

Commit:
update repo to new Scibib version 2

eren-ck reviewed (changes requested):
Please add a License file and simplify the readme.md file. Add ab example figure and link to the scibib pape as a demo page. 

* Why does the `requirements.txt` does not include the lib versions? 
* Remove commented snippets e.g., in `app/backend/db_controller/db.py`
* Is the code secured against SQL injections? 
* Add comments to each method and file. 
* Fix all todos in your code 

Cheers, 
Eren

Commit:
add versions to requirements.txt + remove commented code snippets + comment code + fix todos + update readme (credit libraries and frameworks, add image of homepage) + remove license incompatible libraries

benibaeumle committed:
Create LICENSE

Commit:
add more comments

Commit:
Merge branch 'master' of https://github.com/benibaeumle/SciBib

Commit:
add license header to files

Commit:
add more comments

benibaeumle commented:
I worked through most of the requested changes:

* added versions in requirements.txt
* removed commented snippets from files
* all SQLAlchemy queries and execute statements are implemented as recommended by the docs (https://docs.sqlalchemy.org/en/13/). Nevertheless, the docs do not explicitly state to be secured against SQL injections
* each code snippet has been commented
* ToDOs are fixed
* I checked against the libraries and frameworks used and removed incompatible libraries. All remaining libs are compatible with the GPLv3 license, which I added to the new SciBib version. **Note**: Version 1 was published with Apache 2.0 license

The only remaining issue is to simplify the readme. But I am not sure what this means or how to do it.

Commit:
fix bugs introduced due to recent untested changed

Commit:
update readme

Commit:
add lib to hash passwords

Commit:
add default .env

benibaeumle committed:
Update README.md

benibaeumle committed:
Update README.md

Commit:
fix broken license headers

Commit:
Merge branch 'master' of https://github.com/benibaeumle/SciBib

benibaeumle commented:
All requested changes are implemented now.
